src/ResourceManager/RecoveryServices.Backup/Cmdlets/ProtectionPolicy/SetAzureRmRecoveryServicesPolicy.cs

```diff
@@ -69,9 +69,29 @@ public override void ExecuteCmdlet()
                 }, HydraAdapter);
 
                 IPsBackupProvider psBackupProvider = providerManager.GetProviderInstance(Policy.WorkloadType,
-                                                                                         Policy.BackupManagementType);
-                // now convert hydraPolicy to PSObject
-                WriteObject(psBackupProvider.ModifyPolicy());
+                                                                                         Policy.BackupManagementType);                
+                ProtectionPolicyResponse policyResponse = psBackupProvider.ModifyPolicy();
+
+                if(policyResponse.StatusCode == System.Net.HttpStatusCode.Accepted)
+                {
+                    // Track OperationStatus URL for operation completion
+                    BackUpOperationStatusResponse operationResponse =  WaitForOperationCompletionUsingStatusLink(
+                                                policyResponse.AzureAsyncOperation,
+                                                HydraAdapter.GetProtectionPolicyOperationStatusByURL);
+                                        
+                    if(operationResponse.OperationStatus.Status == OperationStatusValues.Failed.ToString())
+                    {
+                          // if operation failed, then trace warning/error
+                    }
+
+                    // get list of jobIds and return jobResponses                    
+                    WriteObject(GetJobObject(((OperationStatusJobsExtendedInfo)operationResponse.OperationStatus.Properties).JobIds));
```

Comment:
Maybe have a check that JobIds is not null? If response says "Failed" and hasn't created any jobs then we need to log it appropriately.